otwcode/otwarchive
# AO3-5615 Fandom validation rules for unwrangleable tags

# Pull Request Checklist

  as the *first* thing in your pull request title (e.g. `AO3-1234 Fix thing`)
  until they are reviewed and merged before creating new pull requests.

## Issue

https://otwarchive.atlassian.net/browse/AO3-5615

## Purpose

Tags which are fandoms or tags which have no fandoms can no longer be marked unwrangleable. 

Note that I did not use the suggested error message text from the JIRA issue, but instead matched the format of previously existing error messages. 

## Testing Instructions

For this specific feature:
- Log in as tag wrangler
- Find a tag which is a fandom
- Attempt to mark it unwrangleable (note error message informing user fandoms cannot be unwrangleable)
- Find a tag with no fandoms
- Attempt to mark it unwrangleable (note error message informing user that tags with no fandoms cannot be unwrangleable)

However, part of this PR includes a substantial refactor of what was previously a single monster test case:
https://github.com/otwcode/otwarchive/blob/master/features/tags_and_wrangling/tag_wrangling.feature#L26-L256

I've broken it down to be more atomic.  Please double-check that no scenarios have gone missing or just become incorrect.  I've never done tag wrangling so I'm just going by what I see in code base as expected behaviour.

## References

N/A

## Credit

tlee911 (he/him)
